source-control/git.md

```diff
@@ -211,8 +211,76 @@ git stash
 git stash pop
 ```
 
-Or you can simply move the current state into a new branch:
+Or you can move the current state into a new branch:
 
 ```bash
 git stash branch <new_branch_to_save_changes>
 ```
+
+### Working with images, video and other binary content
+
+Avoid committing frequently changed binary files, such as large images, video or compiled code to your git repository. Binary content is not diffed like text content, so cloning or pulling from the repository may pull each revision of the binary file.
+
+One solution to this problem is `Git LFS (Git Large File Storage)` - an open source Git extension for versioning large files. `Git LFS` replaces the binary files with text pointers inside Git, while storing the file contents on a remote server like GitHub.com or Azure DevOps.
+
+#### Benefits of Git LFS
+
+* Uses the end to end Git workflow for all files
+* Git LFS supports file locking to avoid conflicts for undiffable assets
+* Git LFS is fully supported in Azure DevOps Services
+
+#### Limitations of Git LFS
+
+* Everyone contributing to the repository needs to install Git LFS
+* If not set up properly:
+  * Binary files committed through Git LFS are not visible as Git will only download the data describing the large file
+  * Committing large binaries will push the full binary to the repository
+* Git cannot merge the changes from two different versions of a binary file; file locking mitigates this
+* Azure Repos do not support using SSH for repositories with Git LFS tracked files
+
+#### Common commands
+
+Install git lfs
+
+```bash
+git lfs install       # windows
+sudo apt-get git-lfs  # linux
```

Comment:
Nit: May want to quickly look up what the equivalent packages are for `yum` (RedHat/Fedora/CentOS and variants) and `pacman` (ArchLinux and variants) because not everyone will be using Debian/Ubuntu. I think those 3 cover most of the package management for any Linux server distros I'd expect to see currently in the wild.

Reply:
Fixed